deploy/server.deployment.yaml

```diff
@@ -0,0 +1,99 @@
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  namespace: kube-system
+  name: kiam-server
+  labels:
+    app: kiam
+    role: server
+spec:
+  replicas: 3
+  selector:
```

Comment:
I think this is missing:
```
  strategy:
    rollingUpdate:
      maxSurge: 0
      maxUnavailable: 1
    type: RollingUpdate
```
as the helm chart has [here](https://github.com/uswitch/kiam/blob/master/helm/kiam/templates/server-deployment.yaml#L14)

See https://github.com/uswitch/kiam/pull/332/files#r351312455 for background.